alma/lib/Utils/SettingsCustomFields.php

```diff
@@ -39,203 +38,136 @@
 class SettingsCustomFields
 {
-    const SOURCECUSTOMFIELDS = 'SettingsCustomFields';
-
-    /**
-     * Init default custom fileds in ps_configuration table
-     *
-     * @return void
-     */
-    public static function initCustomFields()
-    {
-        $languages = Language::getLanguages(false);
-
-        $keysCustomFields = array_keys(self::customFields());
-        foreach ($keysCustomFields as $keyCustomFields) {
-            Settings::updateValue($keyCustomFields, json_encode(self::getAllLangCustomFieldByKeyConfig($keyCustomFields, $languages)));
-        }
-    }
-
     /**
-     * Default custom fields
+     * Get array custom titles button
      *
      * @return array
      */
-    public static function customFields()
+    public static function getPnxButtonTitle()
     {
-        $module = new Alma();
-        $module->l('Pay in %d installments', self::SOURCECUSTOMFIELDS);
-        $module->l('Pay in %d monthly installments with your credit card.', self::SOURCECUSTOMFIELDS);
-        $module->l('Buy now Pay in %d days', self::SOURCECUSTOMFIELDS);
-        $module->l('Buy now pay in %d days with your credit card.', self::SOURCECUSTOMFIELDS);
-        $module->l('Your cart is not eligible for payments with Alma.', self::SOURCECUSTOMFIELDS);
-        $module->l('At shipping', self::SOURCECUSTOMFIELDS);
-
-        return [
-            PaymentButtonAdminFormBuilder::ALMA_PAYMENT_BUTTON_TITLE => 'Pay in %d installments',
-            PaymentButtonAdminFormBuilder::ALMA_PAYMENT_BUTTON_DESC => 'Pay in %d monthly installments with your credit card.',
-            PaymentButtonAdminFormBuilder::ALMA_DEFERRED_BUTTON_TITLE => 'Buy now Pay in %d days',
-            PaymentButtonAdminFormBuilder::ALMA_DEFERRED_BUTTON_DESC => 'Buy now pay in %d days with your credit card.',
-            ExcludedCategoryAdminFormBuilder::ALMA_NOT_ELIGIBLE_CATEGORIES => 'Your cart is not eligible for payments with Alma.',
-            PaymentOnTriggeringAdminFormBuilder::ALMA_DESCRIPTION_TRIGGER => 'At shipping',
-        ];
+        return CustomFieldsHelper::aggregateAllLanguagesCustomFields(PaymentButtonAdminFormBuilder::ALMA_PNX_BUTTON_TITLE);
     }
 
     /**
-     * Get all languages custom field by key config
+     * Get custom title button by id lang
      *
-     * @param string $keyConfig
-     * @param array $languages
+     * @param int $idLang
      *
-     * @return array
+     * @return string
      */
-    public static function getAllLangCustomFieldByKeyConfig($keyConfig, $languages)
+    public static function getPnxButtonTitleByLang($idLang)
     {
-        foreach ($languages as $language) {
-            $return[$language['id_lang']] = [
-                'locale' => $language['iso_code'],
-                'string' => LocaleHelper::getModuleTranslation(self::customFields()[$keyConfig], self::SOURCECUSTOMFIELDS, $language['iso_code']),
-            ];
-        }
-
-        return $return;
+        return self::getPnxButtonTitle()[$idLang];
```

Comment:
I added a condition to return the default string if $idLang was empty

Reply:
this is not the only one case... it is also possible to get something like

```
if ( ! array_keys_exists( $idLang, self::getPnxButtonTitle() ) ) {
   // then return default string here
}
```

With this test, you will verify both `empty $idLang` and unavailable translation

Reply:
Right, it's more safe. Thanks

Reply:
Sorry to give you picemeal informations... but you should store `self::getPnxButtonTitle()` in a variable because it is potentially called twice into the method and it calls `aggregateAllLanguagesCustomFields()` that will perform requests in database and make complex works.

Reply:
indeed I had not noticed it 🤦‍♂️, it's done